src/main/java/com/minecolonies/coremod/colony/Colony.java

```diff
@@ -442,34 +510,286 @@ public int getID()
+        if (mobKills >= NUM_ACHIEVEMENT_SECOND)
         {
             this.triggerAchievement(ModAchievements.achievementKill25Mobs);
         }
-        if (mobKills >= NUM_MOBS_ACHIEVEMENT_THIRD)
+        if (mobKills >= NUM_ACHIEVEMENT_THIRD)
         {
             this.triggerAchievement(ModAchievements.achievementKill100Mobs);
         }
-        if (mobKills >= NUM_MOBS_ACHIEVEMENT_FOURTH)
+        if (mobKills >= NUM_ACHIEVEMENT_FOURTH)
         {
             this.triggerAchievement(ModAchievements.achievementKill500Mobs);
         }
-        if (mobKills >= NUM_MOBS_ACHIEVEMENT_FIFTH)
+        if (mobKills >= NUM_ACHIEVEMENT_FIFTH)
         {
             this.triggerAchievement(ModAchievements.achievementKill1000Mobs);
         }
     }
+    /**
+     * Increment the ores mined by this colony.
+     * Will award achievements for ores mined.
+     */
+    public void incrementOresMined()
+    {
+        minedOres++;
+        final int Ores = this.getMinedOres();
+        if (Ores >= NUM_ACHIEVEMENT_FIRST)
+        {
+            this.triggerAchievement(ModAchievements.achievementMineOneOre);
+        }
+        if (Ores >= NUM_ACHIEVEMENT_SECOND)
+        {
+            this.triggerAchievement(ModAchievements.achievementMine25Ores);
+        }
+        if (Ores >= NUM_ACHIEVEMENT_THIRD)
+        {
+            this.triggerAchievement(ModAchievements.achievementMine100Ores);
+        }
+        if (Ores >= NUM_ACHIEVEMENT_FOURTH)
+        {
+            this.triggerAchievement(ModAchievements.achievementMine500Ores);
+        }
+        if (Ores >= NUM_ACHIEVEMENT_FIFTH)
+        {
+            this.triggerAchievement(ModAchievements.achievementMine1000Ores);
+        }
+    }
+    /**
+     * Increment the diamonds mined by this colony.
+     * Will award achievements for diamonds mined.
+     */
+    public void incrementDiamondsMined()
+    {
+        minedDiamonds++;
+        final int Diamonds = this.getMinedDiamonds();
+        if (Diamonds >= NUM_ACHIEVEMENT_FIRST)
+        {
+            this.triggerAchievement(ModAchievements.achievementMineOneDiamond);
+        }
+        if (Diamonds >= NUM_ACHIEVEMENT_SECOND)
+        {
+            this.triggerAchievement(ModAchievements.achievementMine25Diamonds);
+        }
+        if (Diamonds >= NUM_ACHIEVEMENT_THIRD)
+        {
+            this.triggerAchievement(ModAchievements.achievementMine100Diamonds);
+        }
+        if (Diamonds >= NUM_ACHIEVEMENT_FOURTH)
+        {
+            this.triggerAchievement(ModAchievements.achievementMine500Diamonds);
+        }
+        if (Diamonds >= NUM_ACHIEVEMENT_FIFTH)
+        {
+            this.triggerAchievement(ModAchievements.achievementMine1000Diamonds);
+        }
+    }
+    /**
+     * Increment the huts built by this colony.
+     * Will award achievements for huts built.
+     */
+    public void incrementHutsBuilt()
+    {
+        builtHuts++;
+        final int Huts = this.getBuiltHuts();
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Rename this local variable name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS00117)